src/backend/distributed/commands/index.c

```diff
@@ -151,6 +151,17 @@ PreprocessIndexStmt(Node *node, const char *createIndexCommand,
 	 * index statements.
 	 */
 	LOCKMODE lockMode = GetCreateIndexRelationLockMode(createIndexStatement);
+
+	/*
+	 * Acquire global lock to prevent concurrent writes or DDL. However,
+	 * we do not bother for CREATE INDEX CONCURRENTLY, since we'll have
+	 * to release the lock.
+	 */
+	if (!createIndexStatement->concurrent)
+	{
+		AcquireDistributedLockOnRelations(list_make1(relationRangeVar), lockMode, 0);
```

Comment:
Should the third parameter, configs,  have DIST_LOCK_REFERENCING_TABLES flag set to acquire any locks?

https://github.com/citusdata/citus/blob/acecdf88278aa4e9fd7d58c0f3c77874ff21ad05/src/backend/distributed/utils/resource_lock.c#L1375